metagpt/provider/anthropic_api.py

```diff
@@ -62,6 +62,9 @@ async def _achat_completion_stream(self, messages: list[dict], timeout: int = US
             elif event_type == "content_block_delta":
                 content = event.delta.text
                 log_llm_stream(content)
+                if self.stream_pipe:
+                    self.stream_pipe.set_message(content)
```

Comment:
The functionality provided by this code `self.stream_pipe.set_message(content)` duplicates that of `log_llm_stream(content)`. You can use the `set_llm_stream_logfunc` method to make `log_llm_stream` call your `stream_pipe.set_message` method.

Reply:
Because `log_llm_stream` uses global logs, it is considered here that web requests will generate log concurrency, resulting in web api data anomalies, so self.stream_pipe is used to ensure that a single request ensures that stream data is not interleaved. And I saw that the `set_llm_stream_logfunc method` is also not suitable for setting global log items.

Reply:
Using context variables (contextvar) can address the concerns you have regarding this issue. Context variables in Python allow you to pass context information in asynchronous code, which is particularly useful for resolving certain concurrency and thread safety issues.

Reply:
Wow, you are right, let me rephrase it